tests/Database/SelectFields/InterfaceTests/InterfaceTest.php

```diff
@@ -78,7 +78,7 @@ public function testGeneratedRelationSqlQuery(): void
 
         $this->assertSqlQueries(<<<'SQL'
 select "id", "title" from "posts";
-select * from "comments" where "comments"."post_id" = ? and "comments"."post_id" is not null order by "comments"."id" asc;
+select "comments"."title", "comments"."post_id" from "comments" where "comments"."post_id" in (?) and "id" >= ? order by "comments"."id" asc;
```

Comment:
It results in the fact that the interface gives the fields that should be selected for the relation instead of selecting `*`, and it allows a custom query to be passed to this relation's query.